psi4/driver/p4util/solvers.py

```diff
@@ -946,15 +976,16 @@ def hamiltonian_solver(engine,
         # Check H2 is PD
         # NOTE: If this triggers failure the SCF solution is not stable. A few ways to handle this
         # 1. Use davidson solver where product function evaluates (H2 * (H1 * X))
-        #    - Poor convergen
+        #    - Poor convergence
         # 2. Switch to CIS/TDA
         #    - User would probably not expect this
         # 3. Perform Stability update and restart with new reference
         if np.any(H2_ss_val < 0.0):
-            raise Exception("H2 is not Positive Definite")
+            msg = ("The H2 matrix is not Positive Definite. " "This means the reference state is not stable.")
+            raise RuntimeError(msg)
 
         # Build H2^(1/2)
-        H2_ss_half = np.dot(H2_ss_vec, np.diag(np.sqrt(H2_ss_val))).dot(H2_ss_vec.T)
+        H2_ss_half = np.einsum("ik,k,jk->ij", H2_ss_vec, np.sqrt(H2_ss_val), H2_ss_vec)
```

Comment:
hooray for einsum notation and readability

Reply:
`np.einsum(..., optimize=True)`